src/Collections/MemoryConfigCollection.php

```diff
@@ -205,35 +200,15 @@ public function getAll()
     }
 
     /**
-     * @deprecated 1.0.0 Use merge() instead
-     *
-     * Synonym for merge()
-     *
      * @param string $class
      * @param string $name
-     * @param mixed  $value
-     * @return $this
-     */
-    public function update($class, $name, $value)
-    {
-        Deprecation::notice('1.0.0', 'Use merge() instead');
-        $this->merge($class, $name, $value);
-        return $this;
-    }
-
-    /**
-     * @param string $class
-     * @param string $name
-     * @param array $value - non-array values are @deprecated 1.12.0
+     * @param array $value
      */
     public function merge($class, $name, $value)
```

Comment:
```suggestion
    public function merge(string $class, string $name, array $value)
```
And delete the docblock `@params` now that we're strong typing this

Reply:
I'll have to update `interface MutableConfigCollectionInterface` as well.

Reply:
I've updated method signature in `interface MutableConfigCollectionInterface` and `class MemoryConfigCollection`